blaze/server/server.py

```diff
@@ -607,21 +584,48 @@ def log_time(start=time()):
 def addserver(payload, serial):
     """Add a data resource to the server.
 
-    The reuest should contain serialized MutableMapping (dictionary) like
-    object, and the server should already be hosting a MutableMapping
-    resource.
+    The request should contain serialized MutableMapping (dictionary) like
+    object, and the server should already be hosting a MutableMapping resource.
     """
+
     data = _get_data.cache[flask.current_app]
 
-    data_not_mm_msg = ("Cannot update blaze server data since its current data"
-                       " is a %s and not a mutable mapping (dictionary like).")
     if not isinstance(data, collections.MutableMapping):
-        return (data_not_mm_msg % type(data), 422)
+        data_not_mm_msg = ("Cannot update blaze server data since its current "
+                           "data is a %s and not a mutable mapping (dictionary "
+                           "like).")
+        return (data_not_mm_msg % type(data), RC.UNPROCESSABLE_ENTITY)
+
+    if not isinstance(payload, collections.Mapping):
+        payload_not_mm_msg = ("Need a dictionary-like payload; instead was "
+                              "given %s of type %s.")
+        return (payload_not_mm_msg % (payload, type(payload)),
+                RC.UNPROCESSABLE_ENTITY)
 
-    payload_not_mm_msg = ("Cannot update blaze server with a %s payload, since"
-                          " it is not a mutable mapping (dictionary like).")
-    if not isinstance(payload, collections.MutableMapping):
-        return (payload_not_mm_msg % type(payload), 422)
+    if len(payload) > 1:
```

Comment:
I blocked this since for now, our only usecases are adding a single dataset at a time.  Generalizing to multiple additions in one `/add` would complicate things, as you point out.

I'm fine generalizing once there's a clear need, but for now, I'm pleading YAGNI :)